src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -91,6 +91,26 @@ namespace
         default: return 0; // default
         }
     }
+
+    boost::optional<BitTorrent::TorrentContentLayout> jsonValueToContentLayout(const QJsonValue &jsonVal)
+    {
+        if (jsonVal.isNull())
```

Comment:
Isn't this redundant since you have the `!isString()` check below?

And what about just the following code?
```c++

boost::optional<BitTorrent::TorrentContentLayout> jsonValueToContentLayout(const QJsonValue &jsonVal)
{
    const QString str = jsonVal.toString();
    if (str.isEmpty())  // this could be redundant too
        return {};
    return Utils::String::parse(str, BitTorrent::TorrentContentLayout::Default);
}
```

Reply:
>Isn't this redundant since you have the` !isString()` check below?

Yes, it's redundant since we don't perform strict validation. Otherwise they're really different cases.

Reply:
>// this could be redundant too

No. Since we need to separate the case when it isn't set at all.

Reply:
>// this could be redundant too

No. Since we need to separate the case when it isn't set at all.

Reply:
>// this could be redundant too

No. Since we need to separate the case when it isn't set at all.